agent_plugin_builder/vendor_dirs.py

```diff
@@ -60,71 +87,135 @@ def check_if_common_vendor_dir_possible(build_dir: Path, uid, gid) -> bool:
 
 
-def generate_windows_vendor_dir(build_dir: Path, uid, gid):
-    client = docker.from_env()
-    commands = (
-        ". /opt/mkuserwineprefix && "
-        f"cd /plugin && wine pip install -r requirements.txt -t src/vendor-windows && "
-        f"chown -R {uid}:{gid} /plugin/src/vendor-windows"
-    )
+def generate_windows_vendor_dir(build_dir: Path, uid: int, gid: int):
+    """
+    Generate the Windows vendor directory by installing the requirements in a Linux Container
+    with Wine installed.
 
-    client.containers.run(
-        PLUGIN_BUILDER_IMAGE,
-        command=f'/bin/bash -c "{commands}"',
-        volumes={build_dir: {"bind": "/plugin", "mode": "rw"}},
-        remove=True,
+    :param build_dir: Path to the build directory.
+    :param uid: User ID to set on the vendor directory.
+    :param gid: Group ID to set on the vendor directory.
+    """
+    command = _build_bash_command(
+        WINDOWS_BUILD_VENDOR_DIR_COMMANDS.format(uid=quote(str(uid)), gid=quote(str(gid)))
     )
+    windows_container = _run_container_with_plugin_dir(PLUGIN_BUILDER_IMAGE, command, build_dir)
+    _log_container_output(windows_container, "Windows vendor directory, ")
+
+
+def _log_container_output(container_logs: bytes, prefix: str = ""):
+    logger.debug(f"{prefix}Container logs: {container_logs.decode('utf-8')}")
 
 
 def generate_requirements_file(build_dir: Path):
+    """
+    Generate the requirements file from the lock file depending on the lock file present.
+
+    :param build_dir: Path to the build directory.
+    """
+    import subprocess
+
+    logger.info("Generating requirements file")
     if (build_dir / "poetry.lock").exists():
         command = ["poetry", "export", "-f", "requirements.txt", "-o", "requirements.txt"]
-        subprocess.check_call(command, cwd=str(build_dir))
+        process = subprocess.Popen(
+            command, cwd=str(build_dir), stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+        )
+        with process.stdout as stdout:  # type: ignore [union-attr]
+            for line in iter(stdout.readline, b""):
+                logger.debug(line)
+
+        return_code = process.wait()
+        if return_code != 0:
+            raise subprocess.CalledProcessError(return_code, command)
     elif (build_dir / "Pipfile.lock").exists():
         command = ["pipenv", "requirements"]
         with (build_dir / "requirements.txt").open("w") as f:
-            subprocess.check_call(command, cwd=str(build_dir), stdout=f)
+            process = subprocess.Popen(
+                command, cwd=str(build_dir), stdout=f, stderr=subprocess.PIPE
+            )
+            with process.stderr as stderr:  # type: ignore [union-attr]
+                for line in iter(stderr.readline, b""):
+                    logger.debug(line)
+
+        return_code = process.wait()
+        if return_code != 0:
+            raise subprocess.CalledProcessError(return_code, command)
+
```

Comment:
We could, perhaps, reduce duplication, here. More importantly, do we need to support pipenv? Since our cookiecutter sets up poetry and doesn't have any option for pipenv, I don't think we really need to support it. I'd prefer we only need to support one (poetry).

Reply:
The argument in favor of supporting pipenv is that all of our current plugins use it. However, if we only use this project to build plugins that we've extracted to their own repositories, and thus use poetry, then we needn't support pipenv

Reply:
That said, I removed it